File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/types/projections/OperationBase.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types.projections;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

/**
 * OperationBase class
 */
@JsonPropertyOrder({"$type"})
public class OperationBase {
    @JsonProperty("$type")
    private String type;
    private String explanation;

    public String getType() {
        return type;
    }

    /**
     * @deprecated This function is extremely likely to be removed in the public interface, and not
     * meant to be called externally at all. Please refrain from using it.
     */
    @Deprecated
    public void setType(final String type) {
        this.type = type;
    }

    public String getExplanation() {
        return explanation;
    }

    public void setExplanation(final String explanation) {
        this.explanation = explanation;
    }
}

File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/CdmFolder/Types/Projections/OperationBase.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder.Types
{
    using Newtonsoft.Json;

    /// <summary>
    /// OperationBase class
    /// </summary>
    public class OperationBase
    {
        [JsonProperty("$type", Order = -2)]
        public string Type { get; internal set; }

        public string Explanation { get; set; }
    }
}
